pkg/api/ref.go

```diff
@@ -79,6 +82,10 @@ func GetReference(obj runtime.Object) (*ObjectReference, error) {
 	}, nil
 }
 
+func mustValidateSelfLink(obj runtime.Object) bool {
+	return reflect.TypeOf(obj) != reflect.TypeOf(&BoundPod{})
+}
+
```

Comment:
@lavalamp Bypassing SelfLink validation for some types seemed like a defensive way to implement this.  Please let me know if you think it safe to remove the empty string comparison entirely.